gst/droidcodec/gstdroidvdec.c

```diff
@@ -908,6 +908,19 @@ gst_droidvdec_configure_state (GstVideoDecoder * decoder, guint width,
   return FALSE;
 }
 
+static inline void
+codec_destroy_fn (DroidMediaCodec * codec)
+{
+  if (!codec)
+    return;
+
+  droid_media_codec_stop (codec);
+  droid_media_codec_drain (codec);
```

Comment:
The original code didn't call `droid_media_codec_drain()` - is that necessary?

Reply:
Yep, calling to this method is needed to fix infinity sleep on cond_variable while the variable is being destroyed. See commit message

Reply:
`droid_media_codec_drain()` is not mentioned in the commit message.

Reply:
and AFAICT `droid_media_codec_stop()` calls `droid_media_codec_drain()` if necessary.